aws_xray_sdk/ext/psycopg2/patch.py

```diff
@@ -0,0 +1,26 @@
+import re
+import wrapt
+
+from aws_xray_sdk.ext.dbapi2 import XRayTracedConn
+
+
+def patch():
+
+    wrapt.wrap_function_wrapper(
+        'psycopg2',
+        'connect',
+        _xray_traced_connect
+    )
+
+
+def _xray_traced_connect(wrapped, instance, args, kwargs):
+
+    conn = wrapped(*args, **kwargs)
+    meta = {
+        'database_type': 'postgresql',
```

Comment:
@haotianw465, how do I see the custom metadata in XRay? I system tested this code by deploying it along with one of my Lambda functions; I did see the correct `database_type`, however I did not see `database_host`, `database_name`, or `database_user` in the Raw Trace Data in the AWS XRay Console.

Reply:
The service will drop unknown keys. Here is a sample sql document the service support.

```json

"sql" : {
     "url": "jdbc:postgresql://aawijb5u25wdoy.cpamxznpdoq8.us-west-2.rds.amazonaws.com:5432/ebdb",
     "preparation": "statement",
     "database_type": "PostgreSQL",
     "database_version": "9.5.4",
     "driver_version": "PostgreSQL 9.4.1211.jre7",
     "user" : "dbuser"
}
```